### ISSUE TITLE
React warning with `theme.components.Modal.defaultProps.overlayProps`

### ISSUE THREAD

**Issue by testerez** [mantinedev/mantine#3731]
### What package has an issue

@mantine/core

### Describe the bug

When setting `Moal.defaultProps.overlayProps` in theme, I get this React warning:
> Warning: React does not recognize the `overlayProps` prop on a DOM element.

Here is my theme:

```ts
export const theme: MantineThemeOverride = {
  components: {
    Modal: {
      defaultProps: {
        size: 640,
        overlayProps: {
          opacity: 0.7,
          blur: 2,
        },
      },
    },
  },
}
```

### What version of @mantine/hooks page do you have in package.json?

@mantine/hooks@6.0.1

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/funny-payne-udkjxy?file=/src/App.tsx

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

**Comment by cyantree** [comment]
@rtivital Is the `{...others}` spread really needed? If yes, for which kind of uses?
https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/ModalBase/ModalBase.tsx#L206
If we can't get rid of it completely is it possible to specify an allow list of supported props?
Otherwise it's quite hard to catch all invalid props and it's only a matter of time until the next issue like this comes up. Or we need to use a library like `filter-invalid-dom-props` that supposedly knows which props are valid and which not.

**Comment by rtivital** [comment]
It is needed to pass additional attributes to the root element, for example data- attributes or event listeners.

**Comment by rtivital** [comment]
Fixed in 6.0.3, thanks @cyantree!
